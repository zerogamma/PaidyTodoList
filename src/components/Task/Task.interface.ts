interface taskProps {
  todo: string;
  position: number;
  id: number;
  remove: (position:number) => void;
  update: (position:number, task:string) => void;
}
