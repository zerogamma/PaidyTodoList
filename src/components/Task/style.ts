import { StyleSheet } from "react-native";

// Utils
import colors from "../../../assets/colors/color";

export const styles = StyleSheet.create({
  item: {
    backgroundColor: colors.secondary,
    padding: 15,
    borderRadius: 10,
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "space-between",
    marginTop: 20,
  },
  todoItemWrapper: {
    flexDirection: "row",
  },
  todoItem: {
    paddingHorizontal: 10,
    flexDirection: "row",
    alignItems: "center",
    flexWrap: "wrap",
  },
  check: {
    width: "80%",
  },
  todoText: {},
  button: {
    color: colors.secondary,
    fontSize: 18,
  },
  iconWrapper: {
    width: 25,
    height: 25,
    backgroundColor: colors.third,
    borderRadius: 30,
    justifyContent: "center",
    alignItems: "center",
  },
});
