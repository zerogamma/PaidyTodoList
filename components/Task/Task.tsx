import React from "react";
import { View, Text, TouchableOpacity } from "react-native";
// Components
import { Ionicons } from "@expo/vector-icons";
// Styles
import { styles } from "./style";

export const Task = ({ todo, position, remove , update}: taskProps) => {
  return (
    <View style={styles.item}>
      <View style={styles.todoItem}>
        <TouchableOpacity style={styles.check}
          onPress={()=> update(position, todo)}
        >
          <Text style={styles.todoText}>{todo}</Text>
        </TouchableOpacity>
      </View>
      <TouchableOpacity
        style={styles.iconWrapper}
        onPress={() => remove(position)}
      >
        <Ionicons name="ios-trash" style={styles.button}></Ionicons>
      </TouchableOpacity>
    </View>
  );
};
