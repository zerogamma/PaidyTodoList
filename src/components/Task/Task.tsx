import React from "react";
import { View, Text, TouchableOpacity } from "react-native";
// Components
import { Ionicons } from "@expo/vector-icons";
// Styles
import { styles } from "./style";

export const Task = ({ todo, id , position, remove, update }: taskProps) => {
  return (
    <View style={styles.item}>
      <View style={styles.todoItem}>
        <TouchableOpacity
          style={styles.check}
          testID="updateTO"
          onPress={() => update(id, todo)} //here is you dont want to use a external data replace with position.
        >
          <Text style={styles.todoText}>{todo}</Text>
        </TouchableOpacity>
      </View>
      <TouchableOpacity
        style={styles.iconWrapper}
        onPress={() => remove(id)} //here is you dont want to use a external data replace with position.
        testID="removeTO"
      >
        <Ionicons name="ios-trash" style={styles.button}></Ionicons>
      </TouchableOpacity>
    </View>
  );
};
