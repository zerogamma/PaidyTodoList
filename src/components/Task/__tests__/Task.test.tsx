import React from "react";
import { Task } from "../Task";
import { render, fireEvent , cleanup} from "@testing-library/react-native";

// Setting default Props per render.
const defaultProps = {
  todo: "test",
  position: 0,
  id: 0,
  remove: () => null,
  update: () => null,
};

const TaskComponent = (customProps: JSX.IntrinsicAttributes & taskProps) => {
  return <Task {...defaultProps} {...customProps} />;
};

describe("<TaskComponent />", () => {
  afterEach(cleanup);
  it("renders correctly", () => {
    const container = render(<TaskComponent {...defaultProps} />);
    expect(container).toMatchSnapshot();
  });
  it("Should trigger Update Task data and verified that the return parameter is the same the props", () => {
    const onTriggerUpdateMock = jest.fn();
    const mockData = {todo: "update data", position:0 , id: 0}
    const { getByTestId } = render(
      <TaskComponent {...mockData} update={onTriggerUpdateMock} remove={()=>null}/>
    );
    const icon = getByTestId("updateTO");
    fireEvent.press(icon);
    expect(onTriggerUpdateMock).toHaveBeenCalled()
    expect(onTriggerUpdateMock.mock.calls[0][1]).toBe(mockData.todo)
    expect(onTriggerUpdateMock.mock.calls[0][0]).toBe(mockData.id)
  });
  it("Should trigger Remove Task data and verified that the return parameter is the same as the prop", () => {
    const onTriggerRemoveMock = jest.fn();
    const mockData = {todo: "update data", position:0 , id:0}
    const { getByTestId } = render(
      <TaskComponent {...mockData} remove={onTriggerRemoveMock} update={()=>null}/>
    );
    const icon = getByTestId("removeTO");
    fireEvent.press(icon);
    expect(onTriggerRemoveMock).toHaveBeenCalled()
    expect(onTriggerRemoveMock.mock.calls[0][0]).toBe(mockData.id)
  });
});
