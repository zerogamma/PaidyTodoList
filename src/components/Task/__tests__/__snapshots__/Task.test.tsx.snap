// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TaskComponent /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "borderRadius": 10,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginTop": 20,
      "padding": 15,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "flexWrap": "wrap",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="updateTO"
    >
      <Text
        style={Object {}}
      >
        test
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#0068f2",
        "borderRadius": 30,
        "height": 25,
        "justifyContent": "center",
        "opacity": 1,
        "width": 25,
      }
    }
    testID="removeTO"
  >
    <Text />
  </View>
</View>
`;
